app/api/attest/diamond-hand/route.ts

```diff
@@ -32,7 +44,18 @@ export async function POST(req: NextRequest) {
   if (await attestedDiamondHands(walletAddress)) {
     return NextResponse.json({ error: 'User has attestation' }, { status: 400 })
```

Comment:
This path will be reached when redis cache is reset but user already attested, right?

In this case, maybe we should set the cache here to avoid re-querying the blockchain?

Reply:
I've left the all the guard checks at the start of the functions, I think it makes sense to group them together and don't think blockchain query is too much overhead.